### PR TITLE
Added PINNBench with Burgers Equation

### PR DESCRIPTION
**Pull request by iamyixuan**
I am not sure what test.py should be doing and the paths within. Please take a look and let me know what needs to be changed. Thanks a lot!

### CODE REVIEW

File: lib/PINNBench/Burgers/hpo.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from deephyper.problem import HpProblem
+from .model import PINN, BurgerSupervisor, sanityCheckData
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:2]: organise imports by: external imports first (`from deephyper ...`) then "local/relative" imports (`from .model ...`)

File: lib/PINNBench/Burgers/hpo.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    lr = config["lr"]
+    alpha = config["alpha"]
+    activation = config["activation"]
+    net = PINN(num_layers, hidden_dim, output_dim, activation)
+    sup = BurgerSupervisor(nu, net, epochs, batch_size, lr, alpha)
+    val_f_loss = sup.train(x, y, val)
+    return -val_f_loss
+
+
+def evaluate(config):
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:34]: where are the testing data? I see training `x, y` and validation `val`. why are they formatted differently `x, y` and `val` ?

File: lib/PINNBench/Burgers/hpo.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    epochs = config["epochs"] 
+    batch_size = config["batch_size"]
+    lr = config["lr"]
+    alpha = config["alpha"]
+    activation = config["activation"] 
+    net = PINN(num_layers, hidden_dim, output_dim, activation)
+    sup = BurgerSupervisor(nu, net, epochs, batch_size, lr, alpha)
+    val_f_loss = sup.train(x, y, val)
+
+
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:50]: should be only 1 empty line at the end of file, check your code format with `black` and `flake8` see: https://deephyper.readthedocs.io/en/develop/developer_guides/dev.html

File: lib/PINNBench/Burgers/model.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
+from math import pi as PI
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:1]: always put imports with `from` after imports with `import`
```python
import ...
from ...
```

File: lib/PINNBench/Burgers/model.py [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+    ic_idx = np.where(x_[:,1]==t[0])[0]
+  
+    u_train_idx = np.concatenate([b_up_idx, b_low_idx, ic_idx]).reshape(-1)
+    x_u_train = x_[u_train_idx]
+    y_u_train = y_[u_train_idx]
+
+    NUM_BC_PT = 500
+    u_idx = np.random.permutation(len(x_u_train))[:NUM_BC_PT]
+    NUM_COL_PT = 2000#int(.1 * len(x_)) # 10% of the points in domain.
+    print("Generating data... total size %d, number of collocation points for trianing %d" % (len(x_), NUM_COL_PT))
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:223]: when using print statements in code, either use the `logging` module which allows to active/deactive printing statements and log outputs in file or in stdout, or add a `verbose=0/1` parameter to your function/objects to allow to activate/deactivate the printing.

File: lib/PINNBench/Burgers/model.py [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+    b_up_idx = np.where(x_[:,0]==x[-1])[0]
+    b_low_idx = np.where(x_[:,0]==x[0])[0]
+    ic_idx = np.where(x_[:,1]==t[0])[0]
+  
+    u_train_idx = np.concatenate([b_up_idx, b_low_idx, ic_idx]).reshape(-1)
+    x_u_train = x_[u_train_idx]
+    y_u_train = y_[u_train_idx]
+
+    NUM_BC_PT = 500
+    u_idx = np.random.permutation(len(x_u_train))[:NUM_BC_PT]
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:221]: using the global random state, please use `np.random.RandomState`

File: lib/PINNBench/Burgers/model.py [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+    NUM_COL_PT = 2000#int(.1 * len(x_)) # 10% of the points in domain.
+    print("Generating data... total size %d, number of collocation points for trianing %d" % (len(x_), NUM_COL_PT))
+    all_idx = np.arange(len(x_)).tolist()
+    col_idx = [i for i in all_idx if (i not in u_train_idx)]
+    f_idx = random.sample(col_idx, NUM_COL_PT)
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:226]: also using the global random state, please stick to one module for random numbers (e.g., `numpy.random` and use `numpy.random.RandomState` to define a local random state.

File: lib/PINNBench/Burgers/model.py [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+    x_f_train = x_[f_idx]
+    # y_f_train = y_[f_idx]
+
+    x_val = x_[col_idx][:2048] # keep a small number for now for testing
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:231]: why are you calling testing data with `val`
training -> `train`
validation -> `val`
testing -> `test`

File: lib/PINNBench/Burgers/model.py [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+
+    # shuffle X_u_train:
+    index = np.arange(0, N_u)
+    random_state.shuffle(index)
+    X_u_train = X_u_train[index, :]
+    
+    # make X_f_train:
+    X_f_train = np.zeros((N_f, 2), dtype=float)
+    for row in range(N_f):
+        x = random.uniform(-1, 1)  # x range
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:310]: accessing global seed from `import random` and confusing because using `np.random` or `random` sometimes, be consistent and use `np.random` only. Same for `pi` and `exp` why not sticking to `numpy`?

File: lib/PINNBench/Burgers/model.py [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+        X_f_train[row, 0] = x 
+        X_f_train[row, 1] = t
+
+    # add the boundary points to the collocation points:
+    X_f_train = np.vstack( (X_f_train, X_u_train) )
+
+    # make u_train
+    u_upper =  np.zeros((N_u//4, 1), dtype=float)
+    u_lower =  np.zeros((N_u//4, 1), dtype=float) 
+    u_zero = -np.sin(np.pi * x_zero)
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/model.py:322]: here using `np.pi` when previously importing `pi` from `math` module, be consistent and stick to `numpy` except if particular technical requirements which should be explained in a inline comment in your code.

File: lib/PINNBench/Burgers/test.py [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import os
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/test.py:1]: This file should contain unit tests for your benchmark. You can remove it for now we will discuss how to improve this later.

**Review by Deathn0t — COMMENTED** [review]
Hello @iamyixuan, I am happy to receive your first PR! I added a bunch of comments. Please take a look and message me or directly reply to the comment if something is not clear. Also checkout this page and try using `black` and `flake8` to improve the quality of your code (both of these checker are mandatory on the main repo `deephyper/deephyper` so it is good to have them installed and working otherwise you will not be able to push any code to the main repo): https://deephyper.readthedocs.io/en/develop/developer_guides/dev.html
Best regards,
Romain.

File: lib/PINNBench/Burgers/hpo.py (hunk 1 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/hpo.py:2]: Done

File: lib/PINNBench/Burgers/hpo.py (hunk 2 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/hpo.py:34]: I made some modifications to the data loading process. Now the train, val, and test, which are dictionaries, are loaded from the function.

File: lib/PINNBench/Burgers/model.py [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+   """
+   Returns the velocity field and distance for the analytical solution
+   """
+
+   # Increments
+   DT = TMAX/(NT-1)
+   DX = XMAX/(NX-1)
+
+   # Initialise data structures
+   import numpy as np
```
[the remark this answers is not in the document]
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:19]: Made the change.

File: lib/PINNBench/Burgers/model.py (hunk 10 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:322]: Made it consistent with `numpy` and used `np.pi`.

File: lib/PINNBench/Burgers/model.py [hunk 13]
```diff
@@ -0,0 +1,329 @@
+from math import pi as PI
+from math import exp as exp
```
[the remark this answers is not in the document]
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:2]: Removed `math` module and made it consistent with `numpy`. All other packages are loaded at the top level.

File: lib/PINNBench/Burgers/model.py (hunk 4 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:1]: Done

File: lib/PINNBench/Burgers/model.py (hunk 5 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:223]: I added a PDE solver for the Burgers equation for a more general data generation process and removed the print line.

File: lib/PINNBench/Burgers/model.py [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+  
+    u_train_idx = np.concatenate([b_up_idx, b_low_idx, ic_idx]).reshape(-1)
+    x_u_train = x_[u_train_idx]
+    y_u_train = y_[u_train_idx]
+
+    NUM_BC_PT = 500
+    u_idx = np.random.permutation(len(x_u_train))[:NUM_BC_PT]
+    NUM_COL_PT = 2000#int(.1 * len(x_)) # 10% of the points in domain.
+    print("Generating data... total size %d, number of collocation points for trianing %d" % (len(x_), NUM_COL_PT))
+    all_idx = np.arange(len(x_)).tolist()
```
[the remark this answers is not in the document]
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:224]: This function is removed.

File: lib/PINNBench/Burgers/model.py (hunk 6 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:221]: Changed every randomization process to using local seed. For example, `rd = np.random.RandomState(0)` and then `rd.uniform(size=n)`.

File: lib/PINNBench/Burgers/model.py (hunk 7 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:226]: Changed to local seed.

File: lib/PINNBench/Burgers/model.py (hunk 8 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/model.py:231]: Now the train, val, and test all being dictionaries have consistent format.

File: lib/PINNBench/Burgers/hpo.py [hunk 15]
```diff
@@ -10,41 +11,45 @@
...
+    train, val, test = get_data(NT=200, NX=128, X_U=1, X_L=-1, T_max=1,
+                                Nu=nu)
+    num_layers = config["num_layers"]
     hidden_dim = config["hidden_dim"]
     output_dim = 1
-    nu = 0.01
-    epochs = config["epochs"] 
-    batch_size = config["batch_size"]
+    epochs = config["epochs"]
+    #batch_size = config["batch_size"]
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/hpo.py:28]: is the batch size used somewhere?  When you are not using variables you can use default parameter values for k,v dictionary access such as `batch_size = config.get("batch_size", 32)`. This will allow to just changing the `problem` definition.

File: lib/PINNBench/Burgers/requirements.txt [hunk 16]
```diff
@@ -1,226 +1,7 @@
...
-xlrd @ file:///home/conda/feedstock_root/build_artifacts/xlrd_1610224409810/work
-xmltodict==0.13.0
-yacs==0.1.8
-yarl @ file:///Users/runner/miniforge3/conda-bld/yarl_1610354163380/work
-yellowbrick @ file:///home/conda/feedstock_root/build_artifacts/yellowbrick_1620328308674/work
-zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1614945704755/work
+phi==0.6.7
+phiflow==2.2.5
+tensorflow==2.10.1
+tensorflow_macos==2.10.0
```
Comment (Deathn0t) [inline lib/PINNBench/Burgers/requirements.txt:7]: so this benchmark is unusable on something else than macOS?

Comment (iamyixuan) [reply lib/PINNBench/Burgers/requirements.txt]: I removed the tf_macos requirement. It should work on other non-mac machines.

File: lib/PINNBench/Burgers/hpo.py (hunk 15 above)
Comment (iamyixuan) [reply lib/PINNBench/Burgers/hpo.py]: Since we train the PINN using two types of training points (boundary and collocation points), and they have different sizes, I took out the batch size. It is not in the search space anymore. We may add it back later if appropriate.

**Review by Deathn0t — APPROVED** [review]
